### PR TITLE
feat(DENG-582): fxa_all_events updated to include all fxa events

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(DENG-582): fxa_all_events updated to include all fxa events

This is part 1 to deprecate the following views:
- `firefox_accounts.fxa_content_auth_events`
- `firefox_accounts.fxa_content_auth_oauth_events`
- `firefox_accounts.fxa_content_auth_stdout_events`

All of these do very similar thing, union different fxa event tables together. It can be quite confusing the the objective is to have a single view to do this to avoid confusion of "which thing should I use".

Related PR links:

- Part 1: https://github.com/mozilla/bigquery-etl/pull/3407
- Part 2: https://github.com/mozilla/bigquery-etl/pull/3408
- Part 3: https://github.com/mozilla/bigquery-etl/pull/3409

----

The next step will be to wait until the new deprecation process in in place and then continue this deprecation with:

- Part 4: Changing the fxa event views to be deprecated to throw an error when queried (comment out the existing logic and add select ERROR(deprecation error))
- Part 5: After a time period (2-4 weeks) completely delete the view files and remove them from BQ

----

### Note 

dry-run currently failing because the updated version of `fxa_all_events` needs to be deployed. Once the changes PR changes have been reviewed I will deploy the new version of the view and the check should pass.

### CODE REVIEW

**Review by relud — APPROVED** [review]

**Comment by kik-kik** [comment]
@sean-rose added you to this PR since some of your dataset logic would be affected by this change (nothing should change). Just wanted to give you a chance to have a look at this before I merge it in case you spot any issues.

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_daily_v1/query.sql [hunk 1]
```diff
@@ -34,18 +34,49 @@ CREATE TEMP FUNCTION udf_contains_registration(
...
+    connect_device_flow,
+    connect_device_os,
+    sync_device_count,
+    sync_active_devices_day,
+    sync_active_devices_week,
+    sync_active_devices_month,
+    email_sender,
+    email_service,
+    email_template,
+    email_version,
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_daily_v1/query.sql:74]:
```suggestion
    `timestamp`,
    event_type,
    user_id,
    app_version,
    os_name,
    os_version,
    country,
    `language`,
    -- cert_signed is specific to sync, but these events do not have the
    -- 'service' field populated, so we fill in the service name for this special case.
    IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service,
```

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/init.sql [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+    country,
+    `language`,
+    entrypoint,
+    utm_term,
+    utm_medium,
+    utm_source,
+    utm_campaign,
+    utm_content,
+    ua_version,
+    ua_browser,
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/init.sql:27]:
```suggestion
    `timestamp`,
    user_id,
    IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service,
    os_name,
    flow_id,
    event_type,
    country,
    entrypoint,
```

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/query.sql [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
+    country,
+    `language`,
+    entrypoint,
+    utm_term,
+    utm_medium,
+    utm_source,
+    utm_campaign,
+    utm_content,
+    ua_version,
+    ua_browser,
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/query.sql:19]:
```suggestion
    `timestamp`,
    user_id,
    IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service,
    os_name,
    flow_id,
    event_type,
    country,
    entrypoint,
```

File: sql/moz-fx-data-shared-prod/mozilla_vpn_derived/funnel_product_page_to_subscribed_v1/query.sql [hunk 4]
```diff
@@ -20,10 +20,65 @@ WITH stripe_plans AS (
...
+    email_sender,
+    email_service,
+    email_template,
+    email_version,
+    plan_id,
+    product_id,
+    promotion_code,
+    payment_provider,
+    checkout_type,
+    source_country,
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/mozilla_vpn_derived/funnel_product_page_to_subscribed_v1/query.sql:77]: * `` `timestamp` receiveTimestamp`` looks like a mistake, as `timestamp` ends up aliased as `receiveTimestamp` and there's no column named `timestamp` in the output, so downstream references to `events.timestamp` will fail.
* A number of columns are being selected twice (e.g. `flow_id`, `entrypoint`).
* A number of columns being selected aren't actually used here (e.g. `logger`, `app_version`, `language`).
```suggestion
    `timestamp`,
    flow_id,
    entrypoint,
    entrypoint_experiment,
    entrypoint_variation,
    utm_term,
    utm_source,
    utm_medium,
    utm_campaign,
    utm_content,
    ua_version,
    ua_browser,
    plan_id,
    promotion_code,
    oauth_client_id,
    event_type,
    os_name,
    os_version,
    country,
    user_id,
    service,
    product_id,
```

File: sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql [hunk 5]
```diff
@@ -3,80 +3,178 @@ CREATE OR REPLACE VIEW
...
+    jsonPayload.fields.user_properties,
+    jsonPayload.fields.event_properties,
+    jsonPayload.fields.device_id,
+  FROM
+    `moz-fx-data-shared-prod.firefox_accounts_derived.fxa_stdout_events_v1`
+),
+unioned AS (
+  SELECT
+    *,
+    'fxa_auth_event' AS event_category,
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql:102]: IMO the `fxa_` prefixes and `_event` suffixes on the event categories are kind of superfluous, and I'd suggest simplifying them to just `auth`, `auth_bounce`, etc.

**Review by sean-rose — APPROVED** [review]
r+wc

The issues in `funnel_product_page_to_subscribed_v1` need to be fixed. All other comments are suggestions.

File: sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql [hunk 6]
```diff
@@ -3,80 +3,178 @@ CREATE OR REPLACE VIEW
...
     jsonPayload.fields.event_type,
-    JSON_VALUE(jsonPayload.fields.event_properties, '$.service') AS service
+    jsonPayload.logger,
+    jsonPayload.fields.user_properties,
+    jsonPayload.fields.event_properties,
+    CAST(NULL AS STRING) AS device_id,
   FROM
     `moz-fx-data-shared-prod.firefox_accounts_derived.fxa_oauth_events_v1`
+),
+stdout AS (
```
[the remark this answers is not in the document]
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql:81]: that's fair. I made the change.

File: sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql [hunk 7]
```diff
@@ -3,80 +3,178 @@ CREATE OR REPLACE VIEW
...
     jsonPayload.fields.event_type,
-    JSON_VALUE(jsonPayload.fields.event_properties, '$.service') AS service
+    jsonPayload.logger,
+    jsonPayload.fields.user_properties,
+    jsonPayload.fields.event_properties,
+    jsonPayload.fields.device_id,
   FROM
     `moz-fx-data-shared-prod.firefox_accounts_derived.fxa_content_events_v1`
 ),
+--
```
[the remark this answers is not in the document]
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql:62]: Added the same comment up here for extra clarity.

File: sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql (hunk 5 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql]: Made the change. 🙇

**Pull request merged**